.pre-commit-config.yaml

```
@@ -0,0 +1,71 @@
+# MIT License
+#
+# Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# Pre-configuration file for pre-commit hooks
+# This is optional. To run pre-commit, you need to install it and run it in the root of your repository.
+
+
+exclude: \.(svg)$ # Exclude files with these extensions
+default_stages: [pre-commit]
+fail_fast: true                        # Stop on the first failed hook.
+repos:
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: check-yaml # Check YAML files for syntax errors
```

```suggestion
      - id: check-yaml # Check YAML files for syntax errors
      - id: trailing-whitespace # Remove trailing whitespace
      - id: end-of-file-fixer # Fix files to have a newline at the end
```
Added a couple other hooks to your config. `clang-format` and `cmake-format` will catch this on the files that they care about, but I noticed the trailing whitespaces on some of the scripts that you modified. These checks would have caught it.